streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -371,6 +372,9 @@ public boolean isStartingRunningOrPartitionAssigned() {
     private volatile KafkaFutureImpl<Uuid> restoreConsumerInstanceIdFuture = new KafkaFutureImpl<>();
     private volatile KafkaFutureImpl<Uuid> producerInstanceIdFuture = new KafkaFutureImpl<>();
 
+    // Missing source topic timeout tracking
+    private long firstMissingSourceTopicTime = -1L;
```

Comment:
Maybe it would make things slighly more easy to read if we'd use 
`org.apache.kafka.common.utils.Timer` for this?

Reply:
Also, can we rename this to a more generic `topicsReadyTimer`? I think we may want to reuse the timer to also time out when internal topics are not created in time.